go/vt/vttablet/tabletmanager/state_change.go

```diff
@@ -125,6 +125,7 @@ func (agent *ActionAgent) broadcastHealth() {
 // refreshTablet needs to be run after an action may have changed the current
 // state of the tablet.
 func (agent *ActionAgent) refreshTablet(ctx context.Context, reason string) error {
+	agent.checkLock()
```

Comment:
I'm wondering if we could have action.lock return a lockHandle (struct {}) that refreshTable requires as input parameter. This will be proof (by compiler) that the function is called while holding the lock. And we avoid maintaining the flag.